internal/check/handler.go

```diff
@@ -329,3 +354,155 @@ func (h *Handler) Check(ctx context.Context, req *rts.CheckRequest) (*rts.CheckR
 		Snaptoken: "not yet implemented",
 	}, nil
 }
+
+// Batch Check Permission Request Parameters
+//
+// swagger:parameters batchCheckPermission
+//
+//lint:ignore U1000 Used to generate Swagger and OpenAPI definitions
+type batchCheckPermission struct {
+	// in: query
+	MaxDepth int `json:"max-depth"`
+
+	// ParallelizationFactor is the maximum number of check requests
+	// that can happen concurrently. Optional. Defaults to 5.
+	//
+	// in: query
+	ParallelizationFactor int `json:"parallelization-factor"`
+
+	// in: body
+	Body batchCheckPermissionBody
+}
+
+// Batch Check Permission Body
+//
+// swagger:model batchCheckPermissionBody
+//
+//lint:ignore U1000 Used to generate Swagger and OpenAPI definitions
+type batchCheckPermissionBody struct {
+	Tuples []*ketoapi.RelationTuple `json:"tuples"`
+}
+
+// Batch Check Permission Result
+//
+// swagger:model batchCheckPermissionResult
+type BatchCheckPermissionResult struct {
+	// An array of check results. The order aligns with the input order.
+	//
+	// required: true
+	Results []*CheckPermissionResultWithError `json:"results"`
+}
+
+// swagger:route POST /relation-tuples/batch/check permission batchCheckPermission
+//
+// # Batch check permissions
+//
+// To learn how relationship tuples and the check works, head over to [the documentation](https://www.ory.sh/docs/keto/concepts/api-overview).
+//
+//	Consumes:
+//	-  application/json
+//
+//	Produces:
+//	- application/json
+//
+//	Schemes: http, https
+//
+//	Responses:
+//	  200: batchCheckPermissionResult
+//	  400: errorGeneric
+//	  default: errorGeneric
+func (h *Handler) batchCheck(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+	results, err := h.doBatchCheck(r.Context(), r.Body, r.URL.Query())
+	if err != nil {
+		h.d.Writer().WriteError(w, r, err)
+		return
+	}
+
+	h.d.Writer().Write(w, r, &BatchCheckPermissionResult{Results: results})
+}
+
+// doBatchCheck is the HTTP entry point for checking batches of tuples
+func (h *Handler) doBatchCheck(ctx context.Context, body io.Reader, query url.Values) ([]*CheckPermissionResultWithError, error) {
+	maxDepth, err := x.GetMaxDepthFromQuery(query)
+	if err != nil {
+		return nil, err
+	}
+	parallelizationFactor := defaultBatchCheckParallelizationFactor
+	if query.Get(parallelizationFactorQueryParam) != "" {
```

Comment:
Sorry for the misunderstanding, but I meant having this as a config parameter in the config (where you also added the max batch size). This is more a parameter the server admin sets to cap the max load of Keto than for a client to set, in my mind.

Reply:
No worries, I've updated it to be a config parameter